editor/modes/visual-editor/block.js

```diff
@@ -133,19 +173,23 @@ class VisualEditorBlock extends wp.element.Component {
 }
 
 export default connect(
-	( state, ownProps ) => ( {
-		order: state.blocks.order.indexOf( ownProps.uid ),
-		block: state.blocks.byUid[ ownProps.uid ],
-		isSelected: state.selectedBlock.uid === ownProps.uid,
-		isHovered: state.hoveredBlock === ownProps.uid,
-		focus: state.selectedBlock.uid === ownProps.uid ? state.selectedBlock.focus : null,
-		isTyping: state.selectedBlock.uid === ownProps.uid ? state.selectedBlock.typing : false,
-	} ),
+	( state, ownProps ) => {
+		const order = state.blocks.order.indexOf( ownProps.uid );
+		return {
+			previousBlock: order === 0 ? null : state.blocks.byUid[ state.blocks.order[ order - 1 ] ],
+			block: state.blocks.byUid[ ownProps.uid ],
+			isSelected: state.selectedBlock.uid === ownProps.uid,
+			isHovered: state.hoveredBlock === ownProps.uid,
+			focus: state.selectedBlock.uid === ownProps.uid ? state.selectedBlock.focus : null,
+			isTyping: state.selectedBlock.uid === ownProps.uid ? state.selectedBlock.typing : false,
+			order
+		};
+	},
 	( dispatch, ownProps ) => ( {
-		onChange( updates ) {
+		onChange( uid, updates ) {
```

Comment:
I like that idea because it promotes consistency, but I don't like that idea because it loses the sense that the component is meant to be specific to a single block.